packages/design-picker/src/components/style-variation-badges/utils.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+	StyleVariation,
+	StyleVariationSettingsColorPalette,
+	StyleVariationStylesColor,
+	StyleVariationPreview,
+	StyleVariationPreviewColorPalette,
+} from '../../types';
+
+const COLOR_PALETTE_SUPPORTS = [ 'background', 'foreground', 'primary', 'secondary', 'tertiary' ];
+
+const GLOBAL_STYLES_SUPPORTS: {
+	style: keyof StyleVariationStylesColor;
+	infix: 'color';
+	mapTo: keyof StyleVariationPreviewColorPalette;
+}[] = [
+	{ style: 'background', infix: 'color', mapTo: 'background' },
+	{ style: 'text', infix: 'color', mapTo: 'foreground' },
+];
+
+function parsePresetDeclaration( preset: string, infix: string ): string | undefined {
+	return preset.match( `--wp--preset--${ infix }--([a-z]+)` )?.[ 1 ];
+}
+
+function getValueFromVariationStylesColor(
+	variation: StyleVariation
+): StyleVariationStylesColor | undefined {
+	return variation.styles?.color;
```

Comment:
Just realized, the endpoint never returns `.styles` 😂 should we fix the endpoint?

Reply:
Added back!